src/components/Form/Widgets/FileWidget.tsx

```diff
@@ -0,0 +1,216 @@
+import UploadFileIcon from '@mui/icons-material/UploadFile';
+import FilePresentIcon from '@mui/icons-material/FilePresent';
+import * as React from 'react';
+import { type DropzoneOptions, useDropzone } from 'react-dropzone';
+import {
+	Box,
+	Link,
+	Typography,
+	CircularProgress,
+	Avatar,
+	InputLabel,
+	List,
+	ListItem,
+	ListItemText,
+} from '@mui/material';
+import type { WidgetProps } from '@rjsf/utils';
+
+export interface OnFileReadSuccessParams {
+	dataUrl: string;
+	uploadedFile: File;
+}
+
+type FileWidgetProps = WidgetProps & {
+	onChange: (params?: OnFileReadSuccessParams) => void;
+	onError?: (errors: string[]) => void;
+	value?: OnFileReadSuccessParams | string;
+};
+
+export const FileWidget = ({
+	onChange,
+	onError,
+	value,
+	...props
+}: FileWidgetProps) => {
+	const [isLoading, setIsLoading] = React.useState<boolean>(false);
+	const [loadingPercentage, setLoadingPercentage] = React.useState<number>(0);
+	const [errors, setErrors] = React.useState<string[]>([]);
+
+	const accept =
+		props.uiSchema?.['ui:options']?.accept ??
+		props.uiSchema?.value?.['ui:options']?.accept;
+	const maxSize =
+		props.uiSchema?.['ui:options']?.maxSize ??
+		props.uiSchema?.value?.['ui:options']?.maxSize;
+
+	const dataUrl = (value: OnFileReadSuccessParams | string) => {
+		return typeof value === 'string' ? value : value.dataUrl;
+	};
+
+	const isImage = (value: OnFileReadSuccessParams | string) => {
+		return dataUrl(value).startsWith('data:image/');
+	};
+
+	const onDrop = React.useCallback(
+		((acceptedFiles, fileRejections) => {
+			setErrors([]);
+
+			if (fileRejections != null && fileRejections.length > 0) {
+				const errs = fileRejections[0].errors.map(
+					({ message }) => `${fileRejections[0].file.name}: ${message}`,
+				);
+				setErrors(errs);
+				onChange();
+				return onError?.(errs);
```

Comment:
I don't see where this is used in the UI PR, could you please clarify the intent here? Why not use `enqueueSnackbar` and remove this function use? (I suspect there may be an issue with ensuring `SnackbarProvider` is used but I expect we should be able to fix it 🤔)

Reply:
@myarmolinsky I can remove the onError indeed. But IMHO the errors showing up as a list on the bottom is better than the snackbar (static for reading the error for a file in case multiple errors are present, for example)

Reply:
I agree you're right, the error list at the bottom indeed looks bettwr than the snackbar, let's stick with it